mountaineer/__tests__/client_builder/test_types.py

```diff
@@ -262,3 +262,55 @@ def test_invalid_literal_values(self, parser):
 
         with pytest.raises(TypeError):
             parser.parse_type(Literal[{"a": 1}])
+
+
+class TestAnnotatedTypes:
+    """Tests for handling typing.Annotated types"""
+
+    @pytest.mark.parametrize(
+        "input_type,expected_type",
+        [
+            (Annotated[str, "metadata"], str),
+            (Annotated[int, "metadata1", "metadata2"], int),
+            (Annotated[List[int], "metadata"], ListOf(int)),
+            (Annotated[str | int, "metadata"], Or(str, int)),
+            (Annotated[Dict[str, int], "metadata"], DictOf(str, int)),
+            (Annotated[str, "metadata"], str),
+            # For nested Annotated types, the inner type is wrapped in Or
+            (List[Annotated[str, "metadata"]], ListOf(Or(str))),
+            (Dict[str, Annotated[int, "metadata"]], DictOf(str, Or(int))),
+            (Annotated[str | None, "metadata"], Or(str, type(None))),
+        ],
+    )
+    def test_annotated_types(self, parser, type_compare, input_type, expected_type):
+        """Test that Annotated types are correctly parsed"""
+        result = parser.parse_type(input_type)
+
+        # For simple types like str, int, the result will be wrapped in Or
+        if not isinstance(expected_type, TypeDefinition):
+            expected_type = Or(expected_type)
+
+        assert type_compare.are_types_equivalent(result, expected_type)
+
+    def test_nested_annotated_types(self, parser, type_compare):
+        """Test nested Annotated types"""
+        # Complex nested type with Annotated
+        complex_type = List[
+            Annotated[Dict[str, Annotated[int | str, "metadata"]], "outer"]
+        ]
+        result = parser.parse_type(complex_type)
+
+        # Expected structure after parsing
+        expected = ListOf(DictOf(key=str, value=Or(int, str)))
+
+        assert type_compare.are_types_equivalent(result, expected)
+
+    def test_annotated_with_no_args(self, parser):
+        """Test that Annotated with no args raises ValueError"""
+        # We can't directly create an Annotated with no args in a type-safe way
+        # So we'll test the error handling by mocking the situation
+        with pytest.raises(ValueError):
+            # Create a mock field_type and origin_type for _parse_origin_type
+            parser._parse_origin_type(Annotated[str, "metadata"], Annotated)
+            # Modify args to be empty to trigger the error
+            parser._parse_origin_type(object(), Annotated)
```

Comment:
Assuming the first `_parse_origin_type` call throws as expected, the code will never get to this line to validate this throws as well.

Should these both be wrapped in separate with pytest.raises contexts?

Reply:
@piercefreeman Good catch. It should be sorted now.